### PR TITLE
Convert channel's date to UTC timezone date

### PR DESCRIPTION
**Pull request by riteshshrv**

### CODE REVIEW

File: sale.py [hunk 1]
```diff
@@ -150,10 +152,19 @@ def get_sale_using_magento_data(cls, order_data):
         else:
             shipment_method = tryton_action['shipment_method']
 
+        datetime_format = '%Y-%m-%d'
```
Comment (priyankarani) [inline sale.py]: use format_datetime http://babel.pocoo.org/docs/api/dates/

File: setup.py [hunk 2]
```diff
@@ -74,6 +74,7 @@ def run(self):
 )
 test_requirements = [
     'mock',
+    'pytz',
 ]
```
Comment (priyankarani) [inline setup.py]: Its not just required for tests

File: sale.py [hunk 3]
```diff
@@ -150,10 +152,19 @@ def get_sale_using_magento_data(cls, order_data):
...
             shipment_method = tryton_action['shipment_method']
 
+        datetime_format = '%Y-%m-%d %H:%M:%S'
+        timezone = pytz.timezone(channel.timezone)
+        sale_date = datetime.strptime(
+            order_data['created_at'], datetime_format
+        )
+        sale_date = timezone.localize(sale_date)
+        utc_sale_date = sale_date.astimezone(pytz.utc)
+        utc_sale_date = utc_sale_date.date().isoformat()
```
Comment (priyankarani) [inline sale.py]: why isoformat ?

File: sale.py [hunk 4]
```diff
@@ -150,10 +152,19 @@ def get_sale_using_magento_data(cls, order_data):
         else:
             shipment_method = tryton_action['shipment_method']
 
+        datetime_format = '%Y-%m-%d %H:%M:%S'
+        timezone = pytz.timezone(channel.timezone)
+        sale_date = datetime.strptime(
+            order_data['created_at'], datetime_format
```
Comment (priyankarani) [inline sale.py]: use format directly here

File: sale.py (hunk 3 above)
Comment (riteshshrv) [inline sale.py]: to convert datetime object into a date string

Comment (priyankarani) [inline sale.py]: but you are saving to date field. why do you need string ?

Comment (riteshshrv) [inline sale.py]: because that's what is required to be used here
https://github.com/fulfilio/trytond-magento/pull/48/files#diff-9cfb5f3dbe3d48ef412a9092fc87311cR167

**Comment by tarunbhardwaj** [comment]
Add test for this.

File: tests/test_sale.py [hunk 5]
```diff
@@ -1398,6 +1412,93 @@ def create_gateways():
...
+        """
+        Tests conversion of date in case user doesnot use timezone
+        field which makes channel's timezone 'UTC' by default.
+        """
+        Sale = POOL.get('sale.sale')
+        Category = POOL.get('product.category')
+
+        with Transaction().start(DB_NAME, USER, CONTEXT):
+            self.setup_defaults()
+            self.import_order_states(self.channel1)
```
Comment (priyankarani) [inline tests/test_sale.py]: you are using channel2 in context but importing states for channel1 ?

File: tests/test_sale.py [hunk 6]
```diff
@@ -1398,6 +1412,93 @@ def create_gateways():
...
+                with Transaction().set_context(company=self.company):
+                    # Create sale order using magento data
+                    with patch(
+                            'magento.Product', mock_product_api(), create=True):
+                        m_sale = Sale.find_or_create_using_magento_data(
+                            order_data
+                        )
+
+                utc_sale_date = get_converted_date(
+                    self.channel2.timezone, order_data['created_at'])
```
Comment (priyankarani) [inline tests/test_sale.py]: assert timezone of channel

File: tests/test_sale.py [hunk 7]
```diff
@@ -1398,6 +1399,55 @@ def create_gateways():
...
+
+                self.assertEqual(
+                    self.channel1.timezone, 'US/Eastern')
+
+                timezone = pytz.timezone(self.channel1.timezone)
+                sale_time = datetime.strptime(
+                    order_data['created_at'], '%Y-%m-%d %H:%M:%S'
+                )
+                sale_time = timezone.localize(sale_time)
+                utc_sale_time = sale_time.astimezone(pytz.utc).date()
```
Comment (tarunbhardwaj) [inline tests/test_sale.py]: utc_sale_date

File: tests/test_sale.py [hunk 8]
```diff
@@ -1398,6 +1399,55 @@ def create_gateways():
...
+                        m_sale = Sale.find_or_create_using_magento_data(
+                            order_data
+                        )
+
+                self.assertEqual(
+                    self.channel1.timezone, 'US/Eastern')
+
+                timezone = pytz.timezone(self.channel1.timezone)
+                sale_time = datetime.strptime(
+                    order_data['created_at'], '%Y-%m-%d %H:%M:%S'
```
Comment (tarunbhardwaj) [inline tests/test_sale.py]: order_data['created_at'] should be an edge case date and test should failing without your patch.

Comment (riteshshrv) [inline tests/test_sale.py]: @tarunbhardwaj tests failed without this patch (as expected)
`======================================================================
FAIL: test_140_check_date_conversion_est_to_utc (tests.test_sale.TestSale)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ritesh/Projects-Active/Fulfil.IO/trytond-magento/tests/test_sale.py", line 1448, in test_140_check_date_conversion_est_to_utc
    '2015-12-22', utc_sale_date
AssertionError: '2015-12-22' != '2015-12-21'`
